pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -416,12 +418,30 @@ func (ctx *context) getDIDDocAndConnection(pubDID string) (*did.Doc, *Connection
 
 	logger.Debugf("creating new '%s' did for connection", didMethod)
 
+	// get the route configs (pass empty service endpoint, as default servie endpoint added in VDRI)
+	serviceEndpoint, routingKeys, err := route.GetRouterConfig(ctx.routeSvc, "")
+	if err != nil {
+		return nil, nil, fmt.Errorf("did doc - fetch router config : %w", err)
+	}
+
 	// by default use peer did
-	newDidDoc, err := ctx.vdriRegistry.Create(didMethod)
+	newDidDoc, err := ctx.vdriRegistry.Create(
+		didMethod,
+		vdri.WithServiceEndpoint(serviceEndpoint),
+		vdri.WithRoutingKeys(routingKeys),
+	)
 	if err != nil {
 		return nil, nil, fmt.Errorf("create %s did: %w", didMethod, err)
 	}
 
+	for _, svc := range newDidDoc.Service {
```

Comment:
Must only select the service with type `did-communication`

Reply:
Use `did.LookUpService`

Reply:
Thanks for pointing it out (I wasn't aware of function). [Updated](https://github.com/hyperledger/aries-framework-go/compare/70e04d120f1905e8fee04847f9fa56eaa14711e8..488523d9b32e5efb07b85bcc1dd8b3965fc9a9e1#diff-86ab318ebdf86972bc415c0a51a2f1ceR437) the code.